gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/V2CanaryController.groovy

```diff
@@ -73,6 +73,20 @@ class V2CanaryController {
                                    configurationAccountName)
   }
 
+  @ApiOperation(value = 'Start a canary execution with the supplied canary config')
+  @RequestMapping(value = '/canary', method = RequestMethod.POST)
+  Map initiateCanaryWithConfig(@RequestBody Map adhocExecutionRequest,
```

Comment:
I am hoping to start seeing `@Valid` tacked on to request bodies so that JSR 30X bean validation occurs and that more issues can fail faster with better error messages.

Now this is Gate and not Kayenta, so this is probably only valid if it is pulling in Kayenta core for its domain models.

Reply:
Related to
- https://github.com/spinnaker/kayenta/issues/354
- https://github.com/spinnaker/kayenta/issues/177

Reply:
NVM, it's just using a generic map, which is unfortunate.

If Kayenta-Core was pulled in for the models and they where used, you could fail faster with `@Valid` and get a better errors, which is probably outside the scope of this PR.

Reply:
Those models would be helpful for spin CLI's `gateapi` also, right now we're just slinging `Map`s around, like you said.